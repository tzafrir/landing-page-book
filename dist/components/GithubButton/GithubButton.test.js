'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _GithubButton = require('./GithubButton');

var _GithubButton2 = _interopRequireDefault(_GithubButton);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step("next", value); }, function (err) { step("throw", err); }); } } return step("next"); }); }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('GithubButton', function () {
  var props = void 0;
  var mounted = void 0;

  var githubButton = function githubButton() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_GithubButton2.default, props));
    }
    return mounted;
  };

  describe('GithubButton', function () {
    beforeEach(function () {
      props = {
        btnType: 'fork',
        btnText: 'Forks',
        username: 'front10',
        repository: 'landing-page-book'
      };
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_GithubButton2.default, props))).toMatchSnapshot();
    });

    it('always renders a span', function () {
      var span = githubButton().find('span');

      expect(span.length).toBeGreaterThan(0);
    });

    it('onchange', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
      var prevProps;
      return regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              prevProps = {
                btnType: 'star',
                btnText: 'Stars',
                username: 'front10',
                repository: 'landing-page-book'
              };
              _context.next = 3;
              return githubButton().instance().onChange(prevProps);

            case 3:

              expect(githubButton().state()).toEqual({
                counter: null,
                showBtnText: true,
                showCounter: true,
                showGithubIcon: false
              });

            case 4:
            case 'end':
              return _context.stop();
          }
        }
      }, _callee, undefined);
    })));
  });
});