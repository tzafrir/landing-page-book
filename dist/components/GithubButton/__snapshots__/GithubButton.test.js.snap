// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubButton GithubButton should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GithubButton
    btnText="Forks"
    btnType="fork"
    className=""
    color="light"
    disabled={false}
    loadingClass="fa fa-circle-o-notch fa-spin"
    repository="landing-page-book"
    rounded={true}
    showBtnText={true}
    showCounter={true}
    showGithubIcon={false}
    username="front10"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          active={false}
          ariaLabel=""
          circle={false}
          className=""
          color="light"
          disabled={false}
          loading={false}
          loadingClass="fa fa-circle-o-notch fa-spin"
          onClick={[Function]}
          outline={false}
          rounded={true}
          size=""
          style={Object {}}
          tooltip=""
          type="button"
        >
          <i
            aria-hidden="true"
            className={undefined}
          />
           
          <span
            className="buttonText"
          >
            Forks
          </span>
        </Button>,
        <Button
          active={false}
          ariaLabel=""
          circle={false}
          className="speech-bubble"
          color="light"
          disabled={false}
          loading={false}
          loadingClass="fa fa-circle-o-notch fa-spin"
          onClick={[Function]}
          outline={false}
          rounded={true}
          size=""
          style={Object {}}
          tooltip=""
          type="button"
        >
          <Icon
            className=""
            icon="fa fa-circle-o-notch fa-spin"
            onClick={[Function]}
            style={Object {}}
            title=""
          />
        </Button>,
      ],
      "className": " GithubDetail_btn_container d-inline",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "ariaLabel": "",
          "children": Array [
            <i
              aria-hidden="true"
              className={undefined}
            />,
            " ",
            <span
              className="buttonText"
            >
              Forks
            </span>,
          ],
          "circle": false,
          "className": "",
          "color": "light",
          "disabled": false,
          "loading": false,
          "loadingClass": "fa fa-circle-o-notch fa-spin",
          "onClick": [Function],
          "outline": false,
          "rounded": true,
          "size": "",
          "style": Object {},
          "tooltip": "",
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Forks",
              "className": "buttonText",
            },
            "ref": null,
            "rendered": "Forks",
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "ariaLabel": "",
          "children": <Icon
            className=""
            icon="fa fa-circle-o-notch fa-spin"
            onClick={[Function]}
            style={Object {}}
            title=""
          />,
          "circle": false,
          "className": "speech-bubble",
          "color": "light",
          "disabled": false,
          "loading": false,
          "loadingClass": "fa fa-circle-o-notch fa-spin",
          "onClick": [Function],
          "outline": false,
          "rounded": true,
          "size": "",
          "style": Object {},
          "tooltip": "",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "icon": "fa fa-circle-o-notch fa-spin",
            "onClick": [Function],
            "style": Object {},
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            active={false}
            ariaLabel=""
            circle={false}
            className=""
            color="light"
            disabled={false}
            loading={false}
            loadingClass="fa fa-circle-o-notch fa-spin"
            onClick={[Function]}
            outline={false}
            rounded={true}
            size=""
            style={Object {}}
            tooltip=""
            type="button"
          >
            <i
              aria-hidden="true"
              className={undefined}
            />
             
            <span
              className="buttonText"
            >
              Forks
            </span>
          </Button>,
          <Button
            active={false}
            ariaLabel=""
            circle={false}
            className="speech-bubble"
            color="light"
            disabled={false}
            loading={false}
            loadingClass="fa fa-circle-o-notch fa-spin"
            onClick={[Function]}
            outline={false}
            rounded={true}
            size=""
            style={Object {}}
            tooltip=""
            type="button"
          >
            <Icon
              className=""
              icon="fa fa-circle-o-notch fa-spin"
              onClick={[Function]}
              style={Object {}}
              title=""
            />
          </Button>,
        ],
        "className": " GithubDetail_btn_container d-inline",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "ariaLabel": "",
            "children": Array [
              <i
                aria-hidden="true"
                className={undefined}
              />,
              " ",
              <span
                className="buttonText"
              >
                Forks
              </span>,
            ],
            "circle": false,
            "className": "",
            "color": "light",
            "disabled": false,
            "loading": false,
            "loadingClass": "fa fa-circle-o-notch fa-spin",
            "onClick": [Function],
            "outline": false,
            "rounded": true,
            "size": "",
            "style": Object {},
            "tooltip": "",
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Forks",
                "className": "buttonText",
              },
              "ref": null,
              "rendered": "Forks",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "ariaLabel": "",
            "children": <Icon
              className=""
              icon="fa fa-circle-o-notch fa-spin"
              onClick={[Function]}
              style={Object {}}
              title=""
            />,
            "circle": false,
            "className": "speech-bubble",
            "color": "light",
            "disabled": false,
            "loading": false,
            "loadingClass": "fa fa-circle-o-notch fa-spin",
            "onClick": [Function],
            "outline": false,
            "rounded": true,
            "size": "",
            "style": Object {},
            "tooltip": "",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "fa fa-circle-o-notch fa-spin",
              "onClick": [Function],
              "style": Object {},
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
